tests/unit/amo/components/TestUserProfile.js

```diff
@@ -896,6 +919,62 @@ describe(__filename, () => {
     sinon.assert.notCalled(dispatchSpy);
   });
 
+  it('returns a 404 if an AddonsByAuthorsCard receives a 404 API error', () => {
+    const { store } = signInUserWithUsername('black-panther');
+    const errorHandler = new ErrorHandler({
+      id: 'some-error-handler-id',
+      dispatch: store.dispatch,
+    });
+
+    const root = renderUserProfile({ errorHandler, store });
+
+    // Simulate an error thrown inside the first AddonsByAuthorsCard.
+    root
+      .find(AddonsByAuthorsCard)
+      .at(0)
+      .prop('errorHandler')
+      .handle(
+        createApiError({
+          response: { status: 404 },
+          apiURL: 'https://some/api/endpoint',
+          jsonResponse: { message: 'not found' },
+        }),
+      );
+
+    // Trigger a re-rendering.
+    root.setProps();
+
+    expect(root.find(NotFound)).toHaveLength(1);
+  });
+
+  it('renders errors from an AddonsByAuthorsCard', () => {
+    const { store } = dispatchSignInActions();
+    const errorHandler = new ErrorHandler({
+      id: 'some-id',
+      dispatch: store.dispatch,
+    });
+
+    const root = renderUserProfile({ errorHandler, store });
+
+    // Simulate an error thrown inside the second AddonsByAuthorsCard.
+    root
+      .find(AddonsByAuthorsCard)
+      .at(1)
```

Comment:
When you have two tests that look the same, it's an indication that you aren't testing the unit of logic in the correct place.

I think you can replace these two tests with one test that makes sure `UserProfile` renders a 404 `NotFound` component given an `errorHandler` with a 404 error. It doesn't matter *where* the error handler gets its error from. The set up could look like:

```js
const { store } = dispatchSignInActions();
const errorHandler = new ErrorHandler({
  id: 'some-id',
  dispatch: store.dispatch,
});
errorHandler.handle(
  createApiError({
    response: { status: 500 },
    apiURL: 'https://some/api/endpoint',
    jsonResponse: { message: 'internal server error' },
  })
);

const root = renderUserProfile({ errorHandler, store });
// Check for the error...
```

The reason you don't need to test the `errorComponent` of each `AddonsByAuthorsCard` is because you already have a test to make sure that their `errorComponent` props are coming from `UserProfile`.

Reply:
right, I don't know what I was thinking.

Reply:
Heh, no worries. It's always easier to spot when doing a code review.